modules/livestore/live_store.go

```diff
@@ -26,6 +28,12 @@ import (
 
 const (
 	liveStoreServiceName = "live-store"
+
+	ringNumTokens = 1 // we only need 1 token in the read ring per instance b/c it's for service discovery only. sharding is done with the parttition ring.
```

Comment:
Can you explain this to my newb brain? I assume its meant to handle ownership in the ring, but how many total tokens are there?

Reply:
i'm not totally convinced this is the right way to do this and have a followup todo to review how mimir uses these rings. what this is doing is creating a ring that is used for discovery only. all we need is a way to map instance name -> instance address.

since we don't actually shard anything with this ring we are adding only 1 token to reduce the overhead of propagating it. i have a hunch there is a "more correct" way to do this which is why i want to dig deeper in the next week or so